examples/nirfsg/attenuation-tables-generation.py

```diff
@@ -0,0 +1,157 @@
+
+
+def raise_if_initialization_error(response):
+    """Raise an exception if an error was returned from Initialize."""
+    if response.status < 0:
+        raise RuntimeError(f"Error: {response.error_message or response.status}")
+    if response.status > 0:
+        sys.stderr.write(f"Warning: {response.error_message or response.status}\n")
+    return response
+
+
+try:
+    response = client.InitWithOptions(
+        nirfsg_types.InitWithOptionsRequest(
+            session_name=SESSION_NAME, resource_name=RESOURCE, option_string=OPTIONS
+        )
+    )
+    raise_if_initialization_error(response)
+    vi = response.vi
+
+    client.ConfigureRefClock(
+        nirfsg_types.ConfigureRefClockRequest(
+            vi=vi,
+            ref_clock_source_mapped=nirfsg_types.REF_CLOCK_SOURCE_ONBOARD_CLOCK,
+            ref_clock_rate=10000000,
+        )
+    )
+
+    client.SetAttributeViString(
+        nirfsg_types.SetAttributeViStringRequest(
+            vi=vi,
+            channel_name="",
+            attribute_id=nirfsg_types.NIRFSG_ATTRIBUTE_REF_CLOCK_SOURCE,
+            value_mapped=nirfsg_types.NIRFSG_STRING_REF_CLOCK_SOURCE_ONBOARD_CLOCK,
+        )
+    )
+
+    raise_if_error(
+        client.ConfigureRF(nirfsg_types.ConfigureRFRequest(vi=vi, frequency=1e9, power_level=-10))
+    )
+
+    # Having a table of attenuations per frequency, we demo how to convert them to S parameters
+    # and download them to RFSG to be automatically set when we change the frequency
+    frequency_table_hz = [1e9, 6e9]
+    attenuation_table_db = [3, 3]
+
+    # Convert to S parameters: Linear attenuation
+    s12 = [math.pow(10, x / 20) for x in attenuation_table_db]
+    # We ned a table that it is frequencies x 2 x 2. Easier to fill it as a 3D matrix
+    sparam = np.zeros(shape=(len(frequency_table_hz), 2, 2))
+    for index in range(len(frequency_table_hz)):
+        sparam[index, 0, 1] = s12[index]
+        sparam[index, 1, 0] = s12[index]
+    # Now flatten into a list
+    sparam = sparam.reshape(np.prod(np.shape(sparam)))
+    sparam = sparam.tolist()
+    # Convert to NI datatype
+    sparam_ni = [nidevice_grpc.NIComplexNumber(real=x, imaginary=0) for x in sparam]
+    # Send them to the instrument
+    client.CreateDeembeddingSparameterTableArray(
+        nirfsg_types.CreateDeembeddingSparameterTableArrayRequest(
+            vi=vi,
+            port="",
+            table_name="myTable",
+            frequencies=frequency_table_hz,
+            sparameter_table=sparam_ni,
+            number_of_ports=2,
+            sparameter_orientation=nirfsg_types.S_PARAMETER_ORIENTATION_PORT1_TOWARDS_DUT,
+        )
+    )
+
+    raise_if_error(client.Initiate(nirfsg_types.InitiateRequest(vi=vi)))
+    print("Generating tone...")
+    # Wait for two seconds and change frequency
+    input("Press any key to stop generation")
+    raise_if_error(client.Abort(nirfsg_types.AbortRequest(vi=vi)))
+except grpc.RpcError as rpc_error:
```

Comment:
This except block should match the others like the [reference-clock.py one ](https://github.com/ni/grpc-device/blob/f8ac15bedf1820c12e37a296e736a1215ba3bbec/examples/nirfsg/reference-clock.py#L94) and include the check for trailing metadata to get any API errors.

Reply:
Done